In [2]:
import pandas as pd

### 1. Read Dataset and Explore the dataset by checking shape, columns, see the first/last 'n' rows using head/tail.

In [3]:
df = pd.read_csv("COVID clinical trials.csv")
df

Rank   NCT Number                                              Title  \
0        1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1        2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2        3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3        4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4        5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
...    ...          ...                                                ...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                  Status         Study Results  \
0      COVID-IDNow  Active, not recruiting  No Results Available   
1         COVID-19      Not yet recruiting  No Results Available   
2      TAC-COVID19              Recruiting  No Results Available   
3         COVID-19  Active, not recruiting  No Results Available   
4     TMF-COVID-19              Recruiting  No Results Available   
...            ...                     ...                   ...   
5778           NaN              Recruiting  No Results Available   
5779           NaN      Not yet recruiting  No Results Available   
5780           NaN              Recruiting  No Results Available   
5781  MindMyMindFU              Recruiting  No Results Available   
5782           NaN      Not yet recruiting  No Results Available   

                                             Conditions  \
0                                               Covid19   
1                                  SARS-CoV-2 Infection   
2                                               covid19   
3                                                 COVID   
4     Maternal Fetal Infection Transmission|COVID-19...   
...                                                 ...   
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5780                               Stress|Psychological   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
0      Diagnostic Test: ID Now™ COVID-19 Screening Test   
1       Drug: Drug COVID19-0001-USR|Drug: normal saline   
2     Other: Lung CT scan analysis in COVID-19 patients   
3             Diagnostic Test: COVID 19 Diagnostic Test   
4     Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   
...                                                 ...   
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   
5779  Other: Antibiotic treatment|Other: No antibiot...   
5780                                   Behavioral: Yoga   
5781                                                NaN   
5782  Behavioral: Brief Cognitive Behavioral Therapy...   

                                       Outcome Measures  \
0     Evaluate the diagnostic performance of the ID ...   
1     Change on viral load results from baseline aft...   
2     A qualitative analysis of parenchymal lung dam...   
3     Proportion of asymptomatic subjects|Proportion...   
4     COVID-19 by positive PCR in cord blood and / o...   
...                                                 ...   
5778  Number of risky drinking days|Number of patien...   
5779  Antibiotic treatment rates in hospitalized chi...   
5780  Perceived Stress|Task switching paradigm|Digit...   
5781  The child's impact of mental health problems r...   
5782  Defense and Veterans Pain Rating Scale (DVPRS)...   

                                  Sponsor/Collaborators  ...  \
0                 Groupe Hospitalier Paris Saint Joseph  ... 

In [ ]:
df.head(5)

In [ ]:
df.shape


In [ ]:
df.columns

In [ ]:
df.head(15)

In [ ]:
df.head(30)

In [ ]:
df.tail(6)

### 2.Extract and document key insights by exploring column data types and applying info and describe and write down your insights inside the markdown.

In [ ]:
df.info()

By applying info() we can see that there are 5783 rows and 27 columns. Futhermore, dtypes:1 float, 1 int, 25 object

In [ ]:
df.describe()

## 3. Loc and iloc
-- Select all columns for the first clinical trial in the dataset.
-- Retrieve the Title and Status of the clinical trial with the NCT Number 'NCT04595136'.
-- Get the Sponsor/Collaborators and Start Date for clinical trials that are Recruiting.
-- Select the first 5 rows and columns Title, Conditions, and Outcome Measures.
-- Find the Completion Date and URL for the last 3 clinical trials in the dataset.

In [ ]:
df.head(1)

In [6]:
df.loc[:,['Rank','NCT Number']]

Rank   NCT Number
0        1  NCT04785898
1        2  NCT04595136
2        3  NCT04395482
3        4  NCT04416061
4        5  NCT04395924
...    ...          ...
5778  5779  NCT04011644
5779  5780  NCT04681339
5780  5781  NCT04740229
5781  5782  NCT04804917
5782  5783  NCT04680000

[5783 rows x 2 columns]

In [7]:
# # -- Select all columns for the first clinical trial in the dataset.
df.iloc[0]

Rank                                                                       1
NCT Number                                                       NCT04785898
Title                      Diagnostic Performance of the ID Now™ COVID-19...
Acronym                                                          COVID-IDNow
Status                                                Active, not recruiting
Study Results                                           No Results Available
Conditions                                                           Covid19
Interventions               Diagnostic Test: ID Now™ COVID-19 Screening Test
Outcome Measures           Evaluate the diagnostic performance of the ID ...
Sponsor/Collaborators                  Groupe Hospitalier Paris Saint Joseph
Gender                                                                   All
Age                                18 Years and older   (Adult, Older Adult)
Phases                                                        Not Applicable

In [7]:
#  -- Retrieve the Title and Status of the clinical trial with the NCT Number 'NCT04595136'.
df.iloc[1]

Rank                                                                       2
NCT Number                                                       NCT04595136
Title                      Study to Evaluate the Efficacy of COVID19-0001...
Acronym                                                             COVID-19
Status                                                    Not yet recruiting
Study Results                                           No Results Available
Conditions                                              SARS-CoV-2 Infection
Interventions                Drug: Drug COVID19-0001-USR|Drug: normal saline
Outcome Measures           Change on viral load results from baseline aft...
Sponsor/Collaborators                             United Medical Specialties
Gender                                                                   All
Age                                18 Years and older   (Adult, Older Adult)
Phases                                                       Phase 1|Phase 2

In [13]:
# -- Get the Sponsor/Collaborators and Start Date for clinical trials that are Recruiting.
data = df[df['Status'] == 'Recruiting']
data.loc[:,["Sponsor/Collaborators","Start Date", "Status"]]

Sponsor/Collaborators         Start Date  \
2                          University of Milano Bicocca        May 7, 2020   
4     Centre Hospitalier Régional d'Orléans|Centre d...        May 5, 2020   
9     Fundacion Arturo Lopez Perez|Confederación de ...      April 7, 2020   
10                                  The Christ Hospital     April 28, 2020   
11                         University of Milano Bicocca     April 16, 2020   
...                                                 ...                ...   
5776  University of Pittsburgh|U.S. National Science...   October 26, 2020   
5777                               University of Zurich    August 25, 2020   
5778  University of Wisconsin, Madison|National Inst...     March 23, 2020   
5780         University of Illinois at Urbana-Champaign  February 10, 2021   
5781  Mental Health Services in the Capital Region, ...     March 22, 2021   

          Status  
2     Recruiting  
4     Recruiting  
9     Recruiting  
10    Recruiting  
11    Recruiting  
...          ...  
5776  Recruiting  
5777  Recruiting  
5778  Recruiting  
5780  Recruiting  
5781  Recruiting  

[2805 rows x 3 columns]

In [14]:
# Select the first 5 rows and columns Title, Conditions, and Outcome Measures.
data1 = df.loc[:,["Title", "Conditions", "Outcome Measures"]]
data1.head()

Title  \
0  Diagnostic Performance of the ID Now™ COVID-19...   
1  Study to Evaluate the Efficacy of COVID19-0001...   
2  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3  The Role of a Private Hospital in Hong Kong Am...   
4         Maternal-foetal Transmission of SARS-Cov-2   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   

                                    Outcome Measures  
0  Evaluate the diagnostic performance of the ID ...  
1  Change on viral load results from baseline aft...  
2  A qualitative analysis of parenchymal lung dam...  
3  Proportion of asymptomatic subjects|Proportion...  
4  COVID-19 by positive PCR in cord blood and / o...

In [15]:
# Find the Completion Date and URL for the last 3 clinical trials in the dataset.
data2 = df.loc[:,["Completion Date","URL"]]
data2.tail(3)

Completion Date                                          URL
5780          July 2021  https://ClinicalTrials.gov/show/NCT04740229
5781  December 31, 2022  https://ClinicalTrials.gov/show/NCT04804917
5782      February 2025  https://ClinicalTrials.gov/show/NCT04680000

### 4.Determine the missing values in the whole dataset and analyze missing values in each column.

In [16]:
df.isnull()

Rank  NCT Number  Title  Acronym  Status  Study Results  Conditions  \
0     False       False  False    False   False          False       False   
1     False       False  False    False   False          False       False   
2     False       False  False    False   False          False       False   
3     False       False  False    False   False          False       False   
4     False       False  False    False   False          False       False   
...     ...         ...    ...      ...     ...            ...         ...   
5778  False       False  False     True   False          False       False   
5779  False       False  False     True   False          False       False   
5780  False       False  False     True   False          False       False   
5781  False       False  False    False   False          False       False   
5782  False       False  False     True   False          False       False   

      Interventions  Outcome Measures  Sponsor/Collaborators  ...  Other IDs  \
0             False             False                  False  ...      False   
1             False             False                  False  ...      False   
2             False             False                  False  ...      False   
3             False             False                  False  ...      False   
4             False             False                  False  ...      False   
...             ...               ...                    ...  ...        ...   
5778          False             False                  False  ...      False   
5779          False             False                  False  ...      False   
5780          False             False                  False  ...      False   
5781           True             False                  False  ...      False   
5782          False             False                  False  ...      False   

      Start Date  Primary Completion Date  Completion Date  First Posted  \
0          False                    False            False         False   
1          False                    False            False         False   
2          False                    False            False         False   
3          False                    False            False         False   
4          False                    False            False         False   
...          ...                      ...              ...           ...   
5778       False                    False            False         False   
5779       False                    False            False         False   
5780       False                    False            False         False   
5781       False                    False            False         False   
5782       False                    False            False         False   

      Results First Posted  Last Update Posted  Locations  Study Documents  \
0                     True               False      False             True   
1                     True               False      False             True   
2                     True               False      False             True   
3                     True               False      False             True   
4                     True               False      False             True   
...                    ...                 ...        ...              ...   
5778                  True               False      False             True   
5779                  True               False      False             True   
5780                  True               False      False             True   
5781                  True               False      False             True   
5782                  True               False      False             True   

        URL  
0     False  
1     False  
2     False  
3     False  
4     False  
...     ...  
5778  False  
5779  False  
5780  False  
5781  False  
5782  False  

[5783 rows x 27 columns]

In [17]:
df.isnull().sum()

Rank                          0
NCT Number                    0
Title                         0
Acronym                    3303
Status                        0
Study Results                 0
Conditions                    0
Interventions               886
Outcome Measures             35
Sponsor/Collaborators         0
Gender                       10
Age                           0
Phases                     2461
Enrollment                   34
Funded Bys                    0
Study Type                    0
Study Designs                35
Other IDs                     1
Start Date                   34
Primary Completion Date      36
Completion Date              36
First Posted                  0
Results First Posted       5747
Last Update Posted            0
Locations                   585
Study Documents            5601
URL                           0
dtype: int64

### 5. Calculate the sum of duplicate rows (5)


In [20]:
df.head().duplicated().sum()

0

### 5. Solve following question by using conditional statements (10) 


<!-- How many studies have an enrollment greater than a certain threshold? --> -->
<!-- How many clinical trials have 'No Results Available'? -->
<!-- How many clinical trials are in an "Completed" and "Recruiting" status? -->
<!-- How many clinical trials are related to 'COVID-19'? -->
<!-- How many clinical trials started after January 1, 2020? -->

In [ ]:
df.head()

In [ ]:
# How many clinical trials have 'No Results Available'?
df["Study Results"].value_counts()

In [ ]:
# How many clinical trials are in an "Completed" and "Recruiting" status?
df[(["Study Results"]=="Has Results") & (df['Status'] == 'Recruiting')].count()


In [ ]:
# How many clinical trials are related to 'COVID-19'?
df.head(2)

In [ ]:
x = df[df["Conditions"] == "Covid19"]

x.count()

In [ ]:
# How many clinical trials started after January 1, 2020?
df[df["Start Date"]>"January 1, 2020"].count()